java/src/jmri/jmrix/openlcb/OlcbReporterManager.java

```diff
@@ -0,0 +1,230 @@
+            if (isLoading) {
+                pendingReporters.add(s);
+            } else {
+                s.finishLoad();
+            }
+        }
+        return s;
+    }
+
+    /**
+     * This function is invoked before an XML load is started. We defer initialization of the
+     * newly created Reporters until finishLoad. This avoids certain quadratic run-time
+     * operations due to update listeners.
+     */
+    public void startLoad() {
+        log.debug("Reporter manager : start load");
+        synchronized (pendingReporters) {
+            isLoading = true;
+        }
+    }
+
+    /**
+     * This function is invoked after the XML load is complete and all Reporters are instantiated
+     * and their feedback type is read in. We use this hook to finalize the construction of the
+     * OpenLCB objects whose instantiation was deferred until the feedback type was known.
+     */
+    public void finishLoad() {
+        log.debug("Reporter manager : finish load");
+        synchronized (pendingReporters) {
+            pendingReporters.forEach(OlcbReporter::finishLoad);
+            pendingReporters.clear();
+            isLoading = false;
+        }
+    }
+
+    @Override
+    @Nonnull
+    public String createSystemName(@Nonnull String curAddress, @Nonnull String prefix) throws JmriException {
+        String tmpPrefix = prefix + typeLetter();
+        String tmpSName  = tmpPrefix + curAddress;
+        try {
+            OlcbAddress.validateSystemNameFormat(tmpSName,Locale.getDefault(),tmpPrefix);
+        }
+        catch ( NamedBean.BadSystemNameException ex ){
+            throw new JmriException(ex.getMessage());
+        }
+        // don't check for integer; should check for validity here
+        return prefix + typeLetter() + curAddress;
+    }
+
+    @Override
+    public boolean allowMultipleAdditions(@Nonnull String systemName) {
+        return true;
+    }
+
+    @Override
+    @Nonnull
+    @javax.annotation.CheckReturnValue
+    public String getNextValidSystemName(@Nonnull NamedBean currentBean) throws JmriException {
+        String currentName = currentBean.getSystemName();
+        return incrementSystemName(currentName);
+    }
+
+    /**
+     * Computes the system name for the next block sensor. This increments the unique ID
+     * of the first 6 components by one.
+     * @param currentName system name for a reporter of a given block
+     * @return next block's system name.
+     */
+    public String incrementSystemName(String currentName) {
+        String oAddr = currentName.substring(getSystemNamePrefix().length());
+        OlcbAddress a = new OlcbAddress(oAddr);
+        // Increments address elements 6 with overflow.
+        int[] e = a.elements();
+        int idx = 5;
+        while(idx > 0) {
+            e[idx]++;
+            if (e[idx] > 255) {
+                e[idx] = 0;
+                --idx;
+            } else {
+                break;
+            }
+        }
```

Comment:
If `idx == 0` after this loop, there is no valid system name. I suggest adding an if statement that checks if idx == 0 and if so, throws an exception. But I'm not entirely sure which exception should be thrown. @bobjacobsen ?

Reply:
The original idea was to never increment the first byte, so for example always stay in the 06.* range. That's what would happen when idx == 0 at the exit.
The result is actually a valid system name, since the rollover from 255 to 0 did happen in all bytes afterwards. It's just that the resulting ID is not numerically larger than what we started with, because the overflow caused a rollover to the beginning of the range. That's a reasonable exit strategy I think.

However, when reviewing this behavior I realized that we actually want to never exit the manufacturer range at all. So I will rather rollover after byte 3 instead of after byte 1.
There are specific examples in the unit test for this code to make this more clear.